nomad/structs/structs.go

```diff
@@ -4054,7 +4054,7 @@ type RecoverableError struct {
 
 // NewRecoverableError is used to wrap an error and mark it as recoverable or
 // not.
-func NewRecoverableError(e error, recoverable bool) error {
+func NewRecoverableError(e error, recoverable bool) *RecoverableError {
```

Comment:
So, I changed this return value from `*RecoverableError` to `error` intentionally in 0.5.3 to avoid turning nil `error`s into non-nil `*RecoverableError`s. Here's an example of that sort of problem: https://play.golang.org/p/Y8z9GdcAhx

So let's keep returning error here to avoid that issue.